### PR TITLE
Use privileged deamonset library

### PR DESCRIPTION
**Pull request by bnshr**

### CODE REVIEW

**Review by sebrandon1 — APPROVED** [review]

**Review by ramperher — CHANGES_REQUESTED** [review]
This change is breaking a lot of tnf tests in our deployments that should be passing, these are now failing or being skipped, please see the last DCI job and its result here: distributed-ci.io/jobs/20e02184-fec4-4a4f-a4f5-f137c5349266/tests

We had issues in the past when dealing with changes in the debug daemonset, so please take care of not breaking its deployment, or if something is needed in the cluster in order to allow the deployment of privileged daemonset, please let us know to do any update in the OCP clusters if required

File: internal/daemonset/daemonset.go [hunk 1]
```diff
@@ -210,7 +35,10 @@ func buildImageWithVersion() string {
 // Deploy daemon set on repo partner
 func DeployPartnerTestDaemonset() error {
 	imageWithVersion := buildImageWithVersion()
-	_, err := CreateDaemonSet(provider.DaemonSetName, provider.DaemonSetNamespace, containerName, imageWithVersion, timeout)
+	oc := clientsholder.GetClientsHolder()
+	k8sPriviledgedDs.SetDaemonSetClient(oc.K8sClient)
+
+	_, err := k8sPriviledgedDs.CreateDaemonSet(provider.DaemonSetName, provider.DaemonSetNamespace, containerName, imageWithVersion, timeout)
```
[the remark this answers is not in the document]
Comment (bnshr) [reply internal/daemonset/daemonset.go]: I [changed the library code!](https://github.com/test-network-function/privileged-daemonset/blob/main/main.go#L29)

File: internal/daemonset/daemonset.go [hunk 2]
```diff
@@ -1,198 +1,23 @@
...
-						},
-						{
-							Effect: "NoSchedule",
-							Key:    "node-role.kubernetes.io/control-plane",
-						},
-					},
-					Volumes: []corev1.Volume{
-						{
-							Name: "host",
-							VolumeSource: corev1.VolumeSource{
```
Comment (edcdavid) [inline internal/daemonset/daemonset.go:150]: The old code was mounting the host file system in /.
We could add this as an option or add "chroot /host" before every command currently executed using the debug pods

Comment (bnshr) [reply internal/daemonset/daemonset.go:150]: For the other places where this is being used, the mounted host file system does not play an important role. So, I would for the time being mount in /.

**Review by ramperher — APPROVED** [review]
Working on DCI side, thanks!

**Review by edcdavid — APPROVED** [review]

**Comment by sebrandon1** [comment]
```
INFO   [Nov  2 13:24:51.310][main.go: 162] Checking if the daemonset exists             
INFO   [Nov  2 13:24:51.343][main.go: 134] Deleting daemonset debug                     
INFO   [Nov  2 13:24:51.469][main.go: 156] Successfully cleaned up daemonset debug      
INFO   [Nov  2 13:24:51.469][main.go: 184] Creating daemonset debug                     
ERROR  [Nov  2 13:24:51.775][daemonset.go: 47] Error deploying partner daemonset object is being deleted: daemonsets.apps "debug" already exists 
--- FAIL: TestTest (0.49s)
FAIL
ERROR  [Nov  2 13:24:51.775][suite_test.go: 156] Error deploying partner daemonset object is being deleted: daemonsets.apps "debug" already exists 
```

Seems like there is something not carried over from the existing code to the library where it keeps hitting this error.

**Pull request merged**